resources/charts/kubernetes/templates/kubelet-ds.yaml

```diff
@@ -0,0 +1,143 @@
+spec:
+  selector:
+    matchLabels:
+      tier: node
+      k8s-app: kubelet
+  template:
+    metadata:
+      labels:
+        tier: node
+        k8s-app: kubelet
+    spec:
+      containers:
+      - name: kubelet
+        image: {{ .Values.kubelet.image }}
+        command:
+        - ./hyperkube
+        - kubelet
+        - --node-ip=$(NODE_PRIVATE_IP)
+        - --anonymous-auth=false
+        - --authentication-token-webhook
+        - --authorization-mode=Webhook
+        - --client-ca-file=/etc/kubernetes/ca.crt
+        - --cluster_dns={{ .Values.kubelet.clusterDNS }}
+        - --cluster_domain={{ .Values.kubelet.clusterDomain }}
+        - --cni-conf-dir=/etc/kubernetes/cni/net.d
+        - --config=/etc/kubernetes/kubelet.config
+        - --kubeconfig=/etc/kubernetes/kubeconfig
+        - --lock-file=/var/run/lock/kubelet.lock
+        - --network-plugin=cni
+        - --node-labels=node.kubernetes.io/node
+        - --pod-manifest-path=/etc/kubernetes/manifests
+        - --read-only-port=0
+        - --volume-plugin-dir=/var/lib/kubelet/volumeplugins
+        env:
+          - name: NODE_PRIVATE_IP
+            valueFrom:
+              fieldRef:
+                fieldPath: status.hostIP
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - mountPath: /var/lib/cni
+          name: coreos-var-lib-cni
+          readOnly: false
+        - mountPath: /var/lib/calico
+          name: coreos-var-lib-calico
+          readOnly: true
+        - mountPath: /opt/cni/bin
+          name: coreos-opt-cni-bin
+          readOnly: true
+        # TODO check if this is needed
+        - name: dev
+          mountPath: /dev
+        - name: run
+          mountPath: /run
+        # TODO check if this is needed
+        - name: sys
+          mountPath: /sys
+          readOnly: true
+        - name: etc-kubernetes
+          mountPath: /etc/kubernetes
+          readOnly: true
+        - name: etc-ssl-certs
+          mountPath: /etc/ssl/certs
+          readOnly: true
+        - name: var-lib-docker
+          mountPath: /var/lib/docker
+        - name: var-lib-kubelet
+          mountPath: /var/lib/kubelet
+          mountPropagation: Bidirectional
+        # TODO check if this is needed
+        # - name: var-lib-rkt
+        #   mountPath: /var/lib/rkt
+        - name: rootfs
+          mountPath: /rootfs
+        - name: logs
+          mountPath: /var/log
+        # TODO /usr/share/certs?
+        # TODO /mnt
+        # TODO /etc/machine-id?
+        # TODO /usr/lib/os-release?
+        # TODO /usr/sbin/iscsiadm?
+      hostNetwork: true
+      hostPID: true
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        operator: Exists
+        effect: NoSchedule
+      volumes:
+      - hostPath:
+          path: /var/lib/cni
+          type: ""
+        name: coreos-var-lib-cni
+      - hostPath:
+          path: /var/lib/calico
+          type: ""
+        name: coreos-var-lib-calico
```

Comment:
We should be consistent about order of keys in those entires. should we move `name` to the top?